app/logic/finder.rb

```diff
@@ -0,0 +1,36 @@
+class Finder
+
+  def initialize options
+    @options = Parser.new(options).perform
+  end
+
+  def perform
+    @query = find_all
+    @query = filter_by_author if options[:author]
+    @query = filter_by_phrase if options[:phrase].present?
+    @query = filter_by_status unless options[:status].nil?
+
+    @query
+  end
+
+  private
+
+  attr_reader :options, :query
+
+  def find_all
+    List.including_items.with_creator
```

Comment:
@vlasiak I suppose it just retuns relation and does not execute the query (fetch data), right? Otherwise let's adjust so the query is being executed one time.